### ISSUE TITLE
AudioWorklet hangs UI when `nodeIntegrationInWorker` is enabled

### ISSUE THREAD

**Issue by samuelmaddock** [electron/electron#25248]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 10.0.1
  * 10.0.0-beta.10
  * 9.2.1
* **Operating System:**
  * Windows 10 (2004)
  * macOS 10.15.6
* **Last Known Working Electron version:**
  * 10.0.0-beta.9
  * Issue is seen in 10.0.0-beta.10

### Expected Behavior
AudioWorklets can be created and destroyed multiple times while `nodeIntegrationInWorker` is enabled.

### Actual Behavior
AudioWorklet can only be created and destroyed once. Upon creating a second AudioWorklet, the UI hangs indefinitely.

### To Reproduce
Electron Fiddle: https://gist.github.com/samuelmaddock/5430479c0a08eca3f693df2518c9887d

1. Load Audio Worklet examples directory: https://googlechromelabs.github.io/web-audio-samples/audio-worklet/
2. Open a demo and click start.
3. Navigate back and return to the demos directory.
4. Open another demo and click start.

UI should no longer be responsive.

**Comment by ckerr** [comment]
I did a little investigating on this. I don't have it fully sussed out yet, but I'm about 2/3s of the way there.

First off, [the patch](https://github.com/electron/electron/pull/24436) that caused this regression -- which, was approved by a couple of people, including me :sweat_smile:  -- introduces a pretty nasty bug. What happens is the worker thread blocks while listening for events in the select() call in `NodeBindingsMac::PollEvents()`. https://github.com/electron/electron/pull/24436/files#diff-561e559fd625c5faf91e1efc111348bcR505 contains a semantic change in `NodeBindings::WakeupEmbedThread()` -- previously, its behavior was "wake up this object's embed thread." The new behavior is "wake up the embed thread unless it's the worker thread. So, the `select()` in `NodeBindingsMac::PollEvents()` never wakes up, and the crash in `uv_walk()` disappears -- *because the worker thread never closes*, so the NodeBindings destructor never finishes destructing, and so `uv_walk()` is never reached. :tada: 

Since the threads never finish, we're leaking every resource associated with the NodeBindings object, including the thread, the uv loop, and so on.

So the next step is to undo that `NodeBindings::WakeupEmbedThread()` change and figure out where the crash is coming from. I think I have that step figured out and will update the ticket shortly with a PR.

**Comment by ckerr** [comment]
I've got a partial fix up in #25332 

This PR wraps the `uv_async_t` objects owned by NodeBindings and ElectronBindings inside a new UvHandle wrapper class which handles `uv_handle_t`s' specific rules about destruction:

> [uv_close()] MUST be called on each handle before memory is released. Moreover, the memory can only be released in close_cb or after it has returned.

The original crash in #22069 was occurring because ElectronBinding's `uv_async_t call_next_tick_async;`was being deleted _immediately_ after being passed to `uv_close()`. It was attached to the NodeBindings' event loop, which is why walking that loop crashed in NodeBindings' destructor -- that `uv_async_t` was still in the loop as a dangling pointer.

The UvHandle wrapper handles the close-delete twostep, fixing the `uv_walk()` crash seen in #22069 in a way that doesn't cause the #25248 regression and which fixes the threads we've been [leaking](https://github.com/electron/electron/issues/25248#issuecomment-687460296). This is all well and good; unfortunately, this now surfaces a second issue:

```
[85352:0904/174418.843359:ERROR:CONSOLE(1995)] "Connection is closed, can't dispatch pending call", source: devtools://devtools/bundled/sdk/sdk.js (1995)
[85359:0904/174418.764382:FATAL:devtools_agent.mojom-blink.cc (http://devtools_agent.mojom-blink.cc)(270)] Check failed: !connected. DevToolsAgent::ReportChildWorkersCallback was destroyed without first either being run or its corresponding binding being closed. It is an error to drop response callbacks which still correspond to an open interface pipe.
0   Electron Framework                  0x000000010a837f09 base::debug::CollectStackTrace(void**, unsigned long) + 9
1   Electron Framework                  0x000000010a738b63 base::debug::StackTrace::StackTrace() + 19
2   Electron Framework                  0x000000010a755445 logging::LogMessage::~LogMessage() + 261
3   Electron Framework                  0x000000010a75613e logging::LogMessage::~LogMessage() + 14
4   Electron Framework                  0x00000001071c7b0e blink::mojom::blink::DevToolsAgent_ReportChildWorkers_ProxyToResponder::OnIsConnectedComplete(bool) + 142
5   Electron Framework                  0x000000010acced01 mojo::(anonymous namespace)::ResponderThunk::IsConnectedAsync(base::OnceCallback<void (bool)>) + 561
6   Electron Framework                  0x00000001071c7a30 std::__1::unique_ptr<blink::mojom::blink::DevToolsAgent_ReportChildWorkers_ProxyToResponder, std::__1::default_delete<blink::mojom::blink::DevToolsAgent_ReportChildWorkers_ProxyToResponder> >::reset(blink::mojom::blink::DevToolsAgent_ReportChildWorkers_ProxyToResponder*) + 112
7   Electron Framework                  0x00000001071c79a9 base::internal::BindState<void (blink::mojom::blink::DevToolsAgent_ReportChildWorkers_ProxyToResponder::*)(), std::__1::unique_ptr<blink::mojom::blink::DevToolsAgent_ReportChildWorkers_ProxyToResponder, std::__1::default_delete<blink::mojom::blink::DevToolsAgent_ReportChildWorkers_ProxyToResponder> > >::Destroy(base::internal::BindStateBase const*) + 25
8   Electron Framework                  0x000000010e1eb39b base::internal::BindState<void (blink::DevToolsAgent::*)(bool, bool, WTF::CrossThreadOnceFunction<void ()>), blink::CrossThreadWeakPersistent<blink::DevToolsAgent>, bool, bool, WTF::CrossThreadOnceFunction<void ()> >::Destroy(base::internal::BindStateBase const*) + 27
9   Electron Framework                  0x000000010e1ea135 blink::DevToolsAgent::IOAgent::ReportChildWorkers(bool, bool, base::OnceCallback<void ()>) + 229
10  Electron Framework                  0x00000001071c3f25 blink::mojom::blink::DevToolsAgentStubDispatch::AcceptWithResponder(blink::mojom::blink::DevToolsAgent*, mojo::Message*, std::__1::unique_ptr<mojo::MessageReceiverWithStatus, std::__1::default_delete<mojo::MessageReceiverWithStatus> >) + 469
11  Electron Framework                  0x000000010e1ea576 blink::mojom::blink::DevToolsAgentStub<mojo::RawPtrImplRefTraits<blink::mojom::blink::DevToolsAgent> >::AcceptWithResponder(mojo::Message*, std::__1::unique_ptr<mojo::MessageReceiverWithStatus, std::__1::default_delete<mojo::MessageReceiverWithStatus> >) + 38
12  Electron Framework                  0x000000010accc0fd mojo::InterfaceEndpointClient::HandleValidatedMessage(mojo::Message*) + 941
13  Electron Framework                  0x000000010accfe9b mojo::MessageDispatcher::Accept(mojo::Message*) + 251
14  Electron Framework                  0x000000010accd681 mojo::InterfaceEndpointClient::HandleIncomingMessage(mojo::Message*) + 97
15  Electron Framework                  0x000000010acd5d4f mojo::internal::MultiplexRouter::ProcessIncomingMessage(mojo::internal::MultiplexRouter::MessageWrapper*, mojo::internal::MultiplexRouter::ClientCallBehavior, base::SequencedTaskRunner*) + 1423
16  Electron Framework                  0x000000010acd5207 mojo::internal::MultiplexRouter::Accept(mojo::Message*) + 407
17  Electron Framework                  0x000000010accfe9b mojo::MessageDispatcher::Accept(mojo::Message*) + 251
18  Electron Framework                  0x000000010acc5296 mojo::Connector::DispatchMessage(mojo::Message) + 470
```

This error message looks straightfoward enough, but I haven't started to track it down yet.

**Comment by codebytere** [comment]
This should be closed by a combination of https://github.com/electron/electron/pull/25332 and the polling PR i just merged!
